apis/telemetry/v1alpha1/shared_types.go

```diff
@@ -68,15 +68,19 @@ type OtlpTLS struct {
 	Key *ValueType `json:"key,omitempty"`
 }
 
+// OtlpOutput OTLP output configuration
+// +kubebuilder:validation:XValidation:rule="((!has(self.path) || size(self.path) <= 0) && (has(self.protocol) && self.protocol == 'grpc')) || (has(self.protocol) && self.protocol == 'http')", message="Path is only available with HTTP protocol"
 type OtlpOutput struct {
-	// Defines the OTLP protocol (http or grpc). Default is GRPC.
+	// Defines the OTLP protocol (HTTP or gRPC). Default is gRPC.
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:default:=grpc
 	// +kubebuilder:validation:Enum=grpc;http
 	Protocol string `json:"protocol,omitempty"`
 	// Defines the host and port (<host>:<port>) of an OTLP endpoint.
 	// +kubebuilder:validation:Required
 	Endpoint ValueType `json:"endpoint"`
+	// Defines OTLP export URL path (only for HTTP protocol)
```

Comment:
I think we should document that we will not add the `/v1/metrics` or `/v1/traces` suffix if the user sets a path.

Reply:
comment updated